### PR TITLE
Add AD role to SP

### PR DESCRIPTION
**Pull request by ravinadhruve10**

### CODE REVIEW

File: modules/services/service-principal/main.tf [hunk 1]
```diff
@@ -25,6 +25,14 @@ resource "azuread_service_principal" "sysdig_sp" {
   }
 }
 
+#---------------------------------------------------------------------------------------------
+# Assign "Directory Reader" AD role to Sysdig SP
+#---------------------------------------------------------------------------------------------
+resource "azuread_directory_role_assignment" "sysdig_ad_reader" {
```
Comment (sysdig-terraform) [inline modules/services/service-principal/main.tf:31]: do we need the same for the organization? probably not since this is AD role.

**Review by sysdig-terraform — APPROVED** [review]

File: modules/services/service-principal/main.tf (hunk 1 above)
Comment (mdkulkarni15) [reply modules/services/service-principal/main.tf:31]: This was from me.

**Review by mdkulkarni15 — APPROVED** [review]

File: modules/services/service-principal/main.tf (hunk 1 above)
Comment (ravinadhruve10) [reply modules/services/service-principal/main.tf:31]: Yes, thats right. The [AD role can scope](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/resources/directory_role_assignment#argument-reference) it to either the application ID or the Directory ID. For tenants, there is a way to set either attribute to = "/"`, to be able to use the role with tenant wide scope. However I am wondering if this wide scope would be a good idea when user gives a specific management group within the tenant.

Comment (ravinadhruve10) [reply modules/services/service-principal/main.tf:31]: Discussed this offline. From the Azure Portal, we observed that by default the AD role assignment takes directory scope type. For now we are moving forward with the default AD role assignment, and can later check how we can limit the scope and if there's a use-case to limit by directory for set of management groups to be onboarded.

**Pull request merged**